samples_project/Assets/SampleViewer/Scripts/SampleSwitcher.cs

```diff
@@ -7,24 +7,49 @@
 using Esri.ArcGISMapsSDK.Components;
 using Esri.HPFramework;
 using System.Collections;
-using System.Collections.Generic;
-using System.IO;
 using UnityEngine;
 using UnityEngine.Rendering;
 using UnityEngine.SceneManagement;
 using UnityEngine.UI;
 
 public class SampleSwitcher : MonoBehaviour
 {
-    public string APIKey = "";
-    public Dropdown PipelineTypeDropdown;
-    public Dropdown SceneDropdown;
+    public GameObject MenuVideo; 
+    private string APIKey;
     public Button ExitButton;
-    public List<string> SceneList = new List<string>();
-    public List<string> PipelineList = new List<string>();
+    private string PipelineModeText;
+    private string SceneText;
     private string PipelineType;
     private string SceneName;
-    private bool EnablePipelineSwitching = true;
+    private int SceneLoadedCount;
+    private Animator animator;
+
+    [SerializeField] private Camera cam;
+    [SerializeField] private Button[] sceneButtons;
+    [SerializeField] private Button[] pipelineButtons;
+
+    private void Start()
+    {
+        animator = GameObject.Find("NotificationMenu").GetComponent<Animator>();
```

Comment:
You should do this with the camera component as well and unserialize it. If the component is private and you can get the it through code you should.

Reply:
Yeah, you don't need to expose the cam field in editor. You can do `GameObject.FindObjectByType<Camera>()` if it's not a child of this game object and the only camera in the scene, `cam = Camera.main.gameObject` if its tagged as the main camera, or `GameObject.Find()`, just to name a few ways to get it

Reply:
What might be the reason why we are not using `SerializedField` on camera object but using `SerializedField` on the ExitButton? From my understanding, `SerializedField` generally is used to save field data, but we are not really saving anything. It's used to to expose private properties on the inspector. Sampleswitcher is acting as a central **GameController**. Why not just make those properties public to improve its efficiency? (we don't have to really worry about exposing these properties).  Finding game objects generally takes longer processing time.

Reply:
Instead of having the buttons be private but in a serializedfield, you should keep them private and unserialize the field and do `FindObjectsOfType()` in order to get them all on `Start()` or `Awake()` found [here](https://docs.unity3d.com/ScriptReference/Object.FindObjectsOfType.html).

Reply:
I usually only use serialize field to debug to ensure that a component is being found properly in the engine if I am getting a null reference exception.

Reply:
As the SampleSwitcher is acting like a GameController, I just don't see the real benefits/point that we "should" keep those properties private.

Reply:
>  From my understanding, SerializedField generally is used to save field data, but we are not really saving anything. 

Ah but you are saving things, you're saving the reference to the `ExitButton` to be used by the script. `Public` and `[SeralizeField] private` do the same thing in editor from a UX perspective, in that they allow for you to expose a field that you can drag and drop objects into (or change a value type, if was say a float value for example)

The big difference is on the coding side: as you already probably know, if an object is marked as `public` ANYTHING that has access to that script can easily make changes to the reference, and change the object to something else, that's a security concern.
The `private` part hides the reference so that other classes can't get at that data/reference and make possibly breaking changes.

Also, another reason is for a cleaner API. If someone decided to programmatically access `SampleSwitcher` for a valid reason, they'll have a cleaner interface to engage with, they won't see `ExitButton` adding to the list of possible things they can interact with, especially since they don't really need to (at this time, maybe we discover a reason to change it in the future, but )

This is a classic oop convention, we want to respect class member encapsulation, which basically means anything that isn't needed outside of a class should not be exposed 

In our case it's really not a big deal because its just a button used by publicly available code, but if you have the ability to easily make your code more secure and cleaner, then you should err on the side of caution and do it.

> Why not just make those properties public to improve its efficiency? (we don't have to really worry about exposing these properties).

`Public` and `[SerializeField] private` take the same amount of time, the reference to the object is serialized once and saved in the unity scene file.

Reply:
> Finding game objects generally takes longer processing time.

True, but more especially in the case if you're making multiple calls to `GameObject.Find()` say every update. If you do it once in `start` or `awake` and cache the result to be used again, it's very performant because you now have a pointer to the object and can easily reuse/interact with it.

Reply:
What might be the reason why we are not using SerializedField on camera object but using SerializedField on the ExitButton?

`SceneViewer` as a concept is the UI manager, the boss of the UI. It owns its children, if you would, so it should have all the references to the UI children it needs to interact with. `[SerializeField]` from a purely UX perspective is a convenience to allow users to drag and drop objects in the editor. (Like I mentioned above there's other things going on with `SerializeField` but that's not necessary for this point)

Typically the camera is not and should not be owned by the `SceneViewer`; `SceneViewer` is a menu, and should only deal with menu things that are inside of it that are necessary for it to work. Objects that exist outside of the `SceneViewer`, like the camera, should ideally be procedurally found and their reference cached for future use.

Honestly though with that said, since the scene is simple enough, we really could keep the cam field as it's not a big deal, but just keep in mind in the future that it's better to procedurally find it.